### PR TITLE
chore(deps): Update deps and move resolutions to root

### PR DESCRIPTION
**Pull request by mayank99**
- Replaced `cpx` with `cpy-cli`. Installed in root package.json because it might be used by multiple packages.
- Also moved `rimraf` to root for the same reason
- Removed `mkdirp` as it's not used anymore
- Moved `resolutions` to root as previously it had no effect
- Went through all `resolutions` and removed unneeded ones

This should help us get rid of the dependabot warnings.

### CODE REVIEW

**Comment by mayank99** [comment]
Looks like `cpy` is not working, even though I thought I tested it. Will move it to draft to investigate tomorrow.

File: packages/iTwinUI-react/package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
     "lint": "eslint \"**/*.{js,ts,tsx}\" --max-warnings=0",
     "lint:fix": "yarn lint --fix && yarn lint:copyright --fix */**/*.{js,ts,tsx}",
     "lint:copyright": "node scripts/copyrightLinter.js",
-    "copy-files": "cpx \"../../{README,LICENSE}.md\" .",
+    "copy-files": "cpy \"../../{README,LICENSE}.md\" .",
```
Comment (bentleyvk) [inline packages/iTwinUI-react/package.json:43]: Try `cpy ../../{README,LICENSE}.md .`

File: package.json [hunk 2]
```diff
@@ -15,9 +15,19 @@
     "lint-staged": "turbo run lint-staged"
   },
   "devDependencies": {
+    "cpy-cli": "^3.1.1",
+    "rimraf": "^3.0.2",
     "turbo": "~1.2.4",
     "webpack": "5"
   },
+  "resolutions": {
```
Comment (veekeys) [inline package.json:23]: List is shorter.
You have removed those which are not needed?

File: packages/iTwinUI-react/package.json [hunk 3]
```diff
@@ -94,10 +93,8 @@
     "jest-junit": "^12.0.0",
     "lint-staged": "^12.1.2",
     "markdown-to-jsx": "6.11.4",
-    "mkdirp": "^1.0.4",
```
Comment (veekeys) [inline packages/iTwinUI-react/package.json:97]: not used anywhere?

File: package.json (hunk 2 above)
Comment (mayank99) [reply package.json:23]: yup

File: packages/iTwinUI-react/package.json (hunk 3 above)
Comment (mayank99) [reply packages/iTwinUI-react/package.json:97]: nope

**Review by veekeys — APPROVED** [review]
hope it works

**Pull request merged**